seqr/views/apis/saved_variant_api.py

```diff
@@ -72,110 +61,180 @@ def create_saved_variant_handler(request):
         family=family,
         saved_variant_json=variant_json
     )
+    return saved_variant
 
-    if non_variant_json.get('note'):
-        _create_variant_note(saved_variant, non_variant_json, request.user)
-    elif non_variant_json.get('tags'):
-        _create_new_tags(saved_variant, non_variant_json, request.user)
 
-    variant_json.update(get_json_for_saved_variant(saved_variant, add_tags=True))
-    return create_json_response({
-        'savedVariantsByGuid': {saved_variant.guid: variant_json},
-    })
+@login_required(login_url=API_LOGIN_REQUIRED_URL)
+@csrf_exempt
+def create_saved_variant_handler(request):
+    variant_json = json.loads(request.body)
+    non_variant_json = {
+        k: variant_json.pop(k, None) for k in ['searchHash', 'tags', 'functionalData', 'notes', 'note', 'submitToClinvar']
+    }
+    family_guid = variant_json.pop('familyGuid')
+    family = Family.objects.get(guid=family_guid)
+    check_permissions(family.project, request.user, CAN_VIEW)
+    # single gene
+    if 'familyGuids' in variant_json.keys():
+        saved_variant = _create_single_saved_variant(variant_json, family)
+        if non_variant_json.get('note'):
+            _create_variant_note([saved_variant], non_variant_json, request.user)
+        elif non_variant_json.get('tags'):
+            _create_new_tags([saved_variant], non_variant_json, request.user)
+        variant_json.update(get_json_for_saved_variant(saved_variant, add_tags=True))
+        return create_json_response({
+            'savedVariantsByGuid': {saved_variant.guid: variant_json},
+        })
+
+    # compound hets
+    else:
+        compound_hets_json = variant_json
```

Comment:
it looks like there is a bunch of copy-paste between this code and the single variant code. Maybe make a helper function?